packages/x-components/src/components/result/base-result-image.vue

```diff
@@ -53,27 +62,60 @@
      */
     @Prop({ required: true })
     protected result!: Result;
+
     /**
      * Animation to use when switching between the placeholder, the loaded image, or the failed
      * image fallback.
      *
      * @public
      */
     @Prop({ default: () => NoElement })
-    public animation!: string | typeof Vue;
+    public loadAnimation!: string | typeof Vue;
+
+    /**
+     * Animation to use when switching between the loaded image and the hover image.
+     *
+     * @public
+     */
+    @Prop()
+    public hoverAnimation!: string | typeof Vue | undefined;
+
+    /**
+     * Copy of the images of the result.
+     *
+     * It is used as a queue of images to load, once an image loads/fails to load, it is removed
+     * from this array.
+     *
+     * @internal
+     */
+    protected images: string[] = this.result.images ?? [];
```

Comment:
As this array changes over time and in the end stores the images pending to be loaded, should we name it `pendingImages` or something similar?